### ISSUE TITLE
Parallelize Erlang compilation

### ISSUE THREAD

**Issue by ericmj** [elixir-lang/elixir#10690]
Today we compile serially due to simply walking over the toposorted list of parse transform and behaviour compile time dependencies. We could do this in parallel by only compiling the dependencies serially and then all dependents in parallel.

EDIT1: We can also do the parsing with `:epp.parse_file/3` in parallel.
EDIT2: Another potential optimizations is to pass the forms from `:epp.parse_file/3` to `:compile.forms/2` instead of `:compile.file/2` to avoid parsing files twice.

**Comment by josevalim** [comment]
Closing in favor of the PR!
